cms/djangoapps/contentstore/views/certificates.py

```diff
@@ -360,39 +369,50 @@ def certificate_data(self):
         return self._certificate_data
 
 
-@login_required
-@require_http_methods(("POST",))
-@ensure_csrf_cookie
-def certificate_activation_handler(request, course_key_string):
+class CertificateActivationAPIView(DeveloperErrorViewMixin, APIView):
     """
-    A handler for Certificate Activation/Deactivation
-
-    POST
-        json: is_active. update the activation state of certificate
+    View for activating or deactivating course certificates.
+    This view allows instructors to toggle the activation state of course certificates.
     """
-    course_key = CourseKey.from_string(course_key_string)
-    store = modulestore()
-    try:
-        course = _get_course_and_check_access(course_key, request.user)
-    except PermissionDenied:
-        msg = _('PermissionDenied: Failed in authenticating {user}').format(user=request.user)
-        return JsonResponse({"error": msg}, status=403)
-
-    data = json.loads(request.body.decode('utf8'))
-    is_active = data.get('is_active', False)
-    certificates = CertificateManager.get_certificates(course)
-
-    # for certificate activation/deactivation, we are assuming one certificate in certificates collection.
-    for certificate in certificates:
-        certificate['is_active'] = is_active
-        break
-
-    store.update_item(course, request.user.id)
-    cert_event_type = 'activated' if is_active else 'deactivated'
-    CertificateManager.track_event(cert_event_type, {
-        'course_id': str(course.id),
-    })
-    return HttpResponse(status=200)
+    permission_classes = [IsAuthenticated]
+
+    @method_decorator(ensure_csrf_cookie)
+    def post(self, request, course_key_string):
+        """
+        A handler for Certificate Activation/Deactivation
+
+        POST
+            json: is_active. update the activation state of certificate
+        """
+        course_key = CourseKey.from_string(course_key_string)
+        serializer = CertificateActivationSerializer(data=request.data)
+
+        store = modulestore()
```

Comment:
move the `modulestore`() call into a separate `get_modulestore()` method within the view. This makes it easier to override or patch in tests.

```
def get_modulestore():
    return modulestore()
````